orc/src/test/java/org/apache/iceberg/orc/TestTableProperties.java

```diff
@@ -0,0 +1,81 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.orc;
+
+import java.io.File;
+import java.util.Map;
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.Files;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableProperties;
+import org.apache.iceberg.hadoop.HadoopTables;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.types.Types;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.mockito.ArgumentCaptor;
+
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+
+public class TestTableProperties {
+
+  public static final Schema SCHEMA = new Schema(
+      Types.NestedField.optional(1, "id", Types.IntegerType.get()),
+      Types.NestedField.optional(2, "data", Types.StringType.get())
+  );
+
+  @ClassRule
+  public static final TemporaryFolder TEMPORARY_FOLDER = new TemporaryFolder();
+
+  @Test
+  public void testAvroProperties() throws Exception {
+    Long stripeSizeBytes = 32L * 1024 * 1024;
+    Long blockSizeBytes = 128L * 1024 * 1024;
+
+    ImmutableMap<String, String> properties = ImmutableMap.of(
+        TableProperties.ORC_STRIPE_SIZE_BYTES, String.valueOf(stripeSizeBytes),
+        TableProperties.ORC_BLOCK_SIZE_BYTES, String.valueOf(blockSizeBytes),
+        TableProperties.DEFAULT_FILE_FORMAT, FileFormat.ORC.name());
+
+    File folder = TEMPORARY_FOLDER.newFolder();
+
+    String warehouse = folder.getAbsolutePath();
+    String tablePath = warehouse.concat("/test");
+    Assert.assertTrue("Should create the table path correctly.", new File(tablePath).mkdir());
+
+    PartitionSpec spec = PartitionSpec.unpartitioned();
+    Table table = new HadoopTables().create(SCHEMA, spec, properties, tablePath);
+
+    ORC.WriteBuilder writeBuilder = spy(ORC.write(Files.localOutput(TEMPORARY_FOLDER.newFile())));
+    writeBuilder.forTable(table);
+    ArgumentCaptor<Map<String, String>> argument = ArgumentCaptor.forClass(Map.class);
+    verify(writeBuilder).setAll(argument.capture());
+    Map<String, String> config = argument.getValue();
+
+    Assert.assertEquals(String.valueOf(stripeSizeBytes), config.get(TableProperties.ORC_STRIPE_SIZE_BYTES));
+    Assert.assertEquals(String.valueOf(blockSizeBytes), config.get(TableProperties.ORC_BLOCK_SIZE_BYTES));
+    Assert.assertEquals(FileFormat.ORC.name(), config.get(TableProperties.DEFAULT_FILE_FORMAT));
```

Comment:
I don't think check the `Map<String, String>` table properties is enough for this PR, because it actually don't cover the the `WriteBuilder.Context::deleteContext` or `WriteBuilder.Context::dataContext` .  the correct approach should be checking the values of `OrcConf.STRIPE_SIZE` & `OrcConf.BLOCK_SIZE` from hadoop configuration.

Reply:
Reasonable. Modified. Thx.